File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTObjectPropertyDef.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTObjectDef } from "./TTObjectDef";
import { TTObjectStateDef } from "./TTObjectStateDef";
import { TTDataType } from "../../DataDictionary/TTDataType";
import { PropertyDefaultValueTypeEnum } from "../../Utils/Enums/PropertyDefaultValueTypeEnum";

export class TTObjectPropertyDef {
    public ObjectDef: TTObjectDef;
    public ObjectDefID: Guid;
    public PropertyDefID: Guid;
    public ShadowOfPropertyDefID: Guid;
    public CodeName: string;
    public Name: string;
    public Description: string;
    public Caption: string;
    public SetScript: string;
    public DataType: TTDataType;
    public DataTypeID: Guid;
    public DefaultValueType: PropertyDefaultValueTypeEnum;
    public DefaultValue: Object;
    public IsDynamic: boolean;
    public IsRequired: boolean;
    public IsProtected: boolean;
    public IsNoLock: boolean;
    //public ParentPropagations: ReadOnlyCollection<TTPropertyPropagationDef>;
    //public ChildPropagations: ReadOnlyCollection<TTPropertyPropagationDef>;
    //public ShadowProperties: ReadOnlyCollection<TTObjectPropertyDef>;
    public IsProtectedOnState(stateDef: TTObjectStateDef): boolean {
        return null;
    }
    public IsCompatibleValue(value: Object): boolean {
        return null;
    }
    public GetSqlLiteral(value: Object): string {
        return null;
    }
    public GetNqlLiteral(value: Object): string {
        return null;
    }
    public ToString(): string {
        return null;
    }
}

File: ATLASDEV/AtlasModel/Enterprise/TTObjectPropertyDef.cs
using AtlasModel.Enterprise.Enums;
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTObjectPropertyDef
    {
        public Guid PropertyDefId { get; set; }
        public Guid ObjectDefId { get; set; }
        public string Name { get; set; }
        public string Caption { get; set; }
        public Guid DataTypeId { get; set; }
        public PropertyDefaultValueTypeEnum DefaultValueType { get; set; }
        public object DefaultValue { get; set; }
        public bool IsDynamic { get; set; }
        public bool IsRequired { get; set; }
        public bool IsProtected { get; set; }
        public string SetScript { get; set; }
        public short CheckOutStatus { get; set; }
        public short IsIndexed { get; set; }
        public Guid? ShadowOfPropertyDefId { get; set; }
        public bool IsNoLock { get; set; }
        public string Description { get; set; }
        public bool AllowUpdateProtected { get; set; }
    }
}